src/deploy/functions/cel.ts

```diff
@@ -0,0 +1,176 @@
+import { FirebaseError } from "../../error";
+import { assertExhaustive } from "../../functional";
+
+export type CelExpression = string;
+export type IdentityExpression = CelExpression;
+export type EqualityExpression = CelExpression;
+export type TernaryExpression = CelExpression;
+
+export type Literal = string | number | boolean;
+type L = "string" | "number" | "boolean";
+
+const identityRegexp = /{{ params\.(\S+) }}/;
+const equalityRegexp = /{{ params\.(\S+) == (.+) }}/;
+const ternaryRegexp = /{{ params\.(\S+) == (.+) ? (.+) : (.+) }/;
```

Comment:
Ternary expressions should not be required to also be an equality expression. It's totally valid to have:

```
const isProd = defineBool("IS_PROD", {
  default: projectId.eq("myProject"),
});

export const myFunc = onRequest({
  minIntsances: isProd.then(10, 0),
}, (req, res) = > res.send("OK"));
```